packages/template-retail-react-app/app/hooks/useBasket.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import {useCustomerBaskets, useCustomerId} from 'commerce-sdk-react'
+
+// This is a simple hook that combined the logic of useCustomerId and useCustomerBaskets to keep the code DRY
+// We can either have this hook later after finishing page integrations or during the integration,
+// this does not have to be a blockers to do page by page integration
+function useBasket(id) {
```

Comment:
@alexvuong I would suggestion that we don't create custom hooks in the project. Reason being: adding these custom hooks increase the complexity of the template and we all know that's a bad thing, it's against the goal of the Extensibility epic.

If we see a common pattern that can simplify the way people use the hook library, we should build that in `commerce-sdk-react`.

Reply:
That sounds like a good idea, we can circle back during one of the integrations 🤔